main.py

```diff
@@ -24,17 +25,130 @@ def setup(self):
         self._flow_mods_sent = OrderedDict()
         self._flow_mods_sent_max_size = FLOWS_DICT_MAX_SIZE
 
+        # object to save and load flows
+        self.storehouse = StoreHouse(self.controller)
+        self.stored_flows = {}
+        self.resent_flows = set()
+
     def execute(self):
         """Run once on NApp 'start' or in a loop.
 
         The execute method is called by the run method of KytosNApp class.
         Users shouldn't call this method directly.
         """
+        self._load_flows()
 
     def shutdown(self):
         """Shutdown routine of the NApp."""
         log.debug("flow-manager stopping")
 
+    @listen_to('kytos/topology.port.created')
+    def resend_stored_flows(self, event):
+        """Resend stored Flows."""
+        dpid = str(event.content['switch'])
+        switch = self.controller.get_switch_by_dpid(dpid)
+        # This can be a problem because this code is running a thread
+        if dpid in self.resent_flows:
+            log.debug(f'Flow already resended to Switch {dpid}')
+            return None
+        if dpid in self.stored_flows:
+            try:
+                flow_list = self.stored_flows[dpid]['flow_list']
+            except KeyError as error:
+                log.info(f'Error to resend stored flow: {error}')
+                return None
```

Comment:
What would generate this KeyError? Either you have a flow_list to install or you don't, right?

Reply:
Yes, you are right. I removed this `try catch` in the last commit.